v2/slogger/rolling_file_appender/rolling_file_appender.go

```diff
@@ -18,59 +18,72 @@
 package rolling_file_appender
 
 import (
-	"fmt"
 	"github.com/mongodb/slogger/v2/slogger"
+
+	"fmt"
 	"os"
 	"path/filepath"
-	"regexp"
 	"sort"
-	"strconv"
 	"sync"
 	"time"
 )
 
 type RollingFileAppender struct {
-	MaxFileSize          int64
-	MaxRotatedLogs       int
+	maxFileSize          int64
+	maxDuration          time.Duration
+	maxRotatedLogs       int
 	file                 *os.File
 	absPath              string
 	curFileSize          int64
 	headerGenerator      func() []string
 	stringWriterCallback func(*os.File) slogger.StringWriter
 	lock                 sync.Mutex
+	state                *state
```

Comment:
This state field is pretty critical for proper behavior (I think it's written correctly) so I wanted to write out my understanding of its lifecycle and help guide some internal documentation for future maintainers.

My guesses:
* `state` will always be instantiated to a non-nil value on a successful `rolling_file_appender.New`
  * If the file exists and we are appending to it, `loadState` is called, initializing `state`
  * If the file does not exist, `Rotate` is called which will initialize `state` to a non-nil value
* `state` is only managing the start timestamp. The current file size is called directly on the file object (for appending) and managed on the `RollingFileAppender` struct directly.
* Aside from appender initialization, a `state` object is only destroyed/created everytime the log file is rotated (conveniently in the `Rotate` method). This requires locking in the write path of `state` and consequently the read path of `state. This is currently accomplished as both `Append` and `Rotate` acquire the `RollingFileAppender` lock.
  * `state` is not accessed outside of those two methods.

Can you confirm (or correct) my understanding? The next step would be to shape this into concise documentation.